### PR TITLE
Allow configuration from secrets manager

### PR DESCRIPTION
**Pull request by julienduchesne**
Here's a summary of what this does. Instead of treating SSM and files differently, they all go through the same process. Configs are aggregated in a a list and are applied in that order:
1. SecretsManager (tgf-config) or SSM (/default/tgf)
2. .tgf.user.config
3. .tgf.config
Applying in that order means that .tgf.config will overwrite .tgf.user.config and SSM params. After that process, CLI arguments are applied and they override all. Behavior seems ok, but I will be adding unit tests before merging, this is a WIP

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -128,178 +89,72 @@ func (config *TGFConfig) InitAWS(profile string) error {
 }
 
 // SetDefaultValues sets the uninitialized values from the config files and the parameter store
+// Priorities (Higher overwrites lower values):
+// 1. SSM Parameter Config
+// 2. Secrets Manager Config (If exists, will not check SSM)
+// 3. .tgf.user.config
```
Comment (jocgir) [inline config.go]: It is not hidden, its tgf.user.config

I should override .tgf.config

Comment (jocgir) [reply config.go]: From what I understand, you reversed the way of evaluating precedence. We should also reverse the evaluation of config files.

Comment (julienduchesne) [reply config.go]: I reversed it but the logic is different. 
You were setting values only if they were unset: That way you set explicit variables and they are not set by other methods
What I do is Unmarshal every layer on top of the others, that way later calls will overwrite values previously set.

**Comment by julienduchesne** [comment]
I'm done! You can review now.

**Pull request merged**